### PR TITLE
Added random_eps parameter

### PR DESCRIPTION
**Pull request by henry-eigen**
Signed-off-by: Henry Eigen <eigen.henry@gmail.com>

# Description

Added random epsilon parameter for use in adversarial training. When true, the attack selects epsilon randomly from a truncated normal distribution as suggested by Alexey Kurakin and Ian J. Goodfellow in "Adversarial Machine Learning at Scale" (https://arxiv.org/pdf/1611.01236.pdf). Attempt number two.

## Improvement

Please check all relevant options.

- [ ] Bug fix (non-breaking)
- [X] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

**Test Configuration**:
- Python 3.6.8
- Keras 2.2.4    

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by beat-buesser** [comment]
Hi @henry-eigen Thank you very much for using ART and your pull request!

**Comment by henry-eigen** [comment]
I changed the eps_step to preserve the initial ratio since you are right that the 1.2 number was arbitrary. I also added the requested documentation info. To prevent the issue of inefficiency when eps is near 0, the code could be changed to include something along the lines of "iters = min(eps + 4, 1.25*eps)" as suggested in the paper for iterative fgsm. Anecdotally, I can say that the issue of generalizing to multiple epsilons is present in PGD training as well as FGSM training. I chose PGD since the literature suggests it is the best for robustness. I appreciate the feedback!

File: art/attacks/evasion/projected_gradient_descent.py [hunk 1]
```diff
@@ -114,6 +125,11 @@ def generate(self, x, y=None, **kwargs):
         adv_x_best = None
         rate_best = None
 
+        if self.random_eps:
+            ratio = self.eps_step / self.eps
+            self.eps = np.round(self.norm_dist.rvs(1)[0], 10)
```
Comment (beat-buesser) [inline art/attacks/evasion/projected_gradient_descent.py]:
```suggestion
            eps = np.round(self.norm_dist.rvs(1)[0], 10)
```

File: art/attacks/evasion/projected_gradient_descent.py [hunk 2]
```diff
@@ -114,6 +125,11 @@ def generate(self, x, y=None, **kwargs):
...
+            self.eps_step = ratio * self.eps
```
Comment (beat-buesser) [inline art/attacks/evasion/projected_gradient_descent.py]:
```suggestion
            eps_step = ratio * self.eps
        else:
            eps = self.eps
            eps_step = self.eps_step
```

Comment (beat-buesser) [reply art/attacks/evasion/projected_gradient_descent.py]: In addition to changing line 137 from
```
 adv_x = self._compute(adv_x, x, targets, self.eps, self.eps_step, self._project,
```
to 
```
adv_x = self._compute(adv_x, x, targets, eps, eps_step, self._project,
```

**Comment by beat-buesser** [comment]
@henry-eigen Thank you for your first contribution to ART!
@minhitbk and @mathsinn Thank you for your reviews!

**Pull request merged**